src/MyParcelComApi.php

```diff
@@ -817,16 +814,34 @@ protected function getResourceUri($resourceType, $id = null)
      * @param array $array
      * @return array
      */
-    private function arrayToFilter(array $array)
+    private function arrayToFilters(array $array)
     {
         $filters = [];
         foreach ($array as $name => $value) {
-            $filters["filter[$name]"] = $value;
+            list($name, $value) = $this->arrayToFilter([$name], $value);
+            $filters["filter".$name] = $value;
         }
 
         return $filters;
     }
 
+    /**
+     * Converts given array to a filter string for the query params.
+     *
+     * @param array $keys
+     * @param $values
+     * @return array
+     */
+    private function arrayToFilter($keys, $values)
```

Comment:
What are these `keys` and `values` that we're expecting in this method? I can't really follow what's happening.